src/main/java/synapseawsconsolelogin/Auth.java

```diff
@@ -264,6 +264,11 @@ String getConsoleLoginURL(HttpServletRequest req, Credentials federatedCredentia
 		return Jwts.parser().parseClaimsJwt(unsignedToken);
 	}
 	
+	// tag values must adhere to the regex: [\p{L}\p{Z}\p{N}_.:/=+\-@]*
```

Comment:
could you please explain why it's limited to these characters?  I'm wondering whether it would be better to handle this at the root (in synapse) rather than downstream?

Reply:
> could you please explain why it's limited to these characters?
Yes, this limit is imposed by AWS.  It is explicit in the returned error message.  It's also documented here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions

> I'm wondering whether it would be better to handle this at the root (in synapse)
The existing Synapse user metadata is not meant to be used only as AWS tags so I would not add such constraints to those fields.  For example the "company" field in the Synapse user profile is not created just to be an AWS tag.  If we wanted to store AWS specific meta data for Synapse users and then use those metadata as AWS resource tags then I would agree to add the tag value constraints to those metadata.  But those would be additional metadata and might not even be stored in Synapse itself.